pom.xml

```diff
@@ -280,6 +282,66 @@
                     </excludes>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-checkstyle-plugin</artifactId>
+                <dependencies>
+                    <dependency>
+                        <groupId>com.puppycrawl.tools</groupId>
+                        <artifactId>checkstyle</artifactId>
+                        <version>8.41.1</version>
+                    </dependency>
+                </dependencies>
+                <!-- Version specified in parent POM -->
+                <configuration>
```

Comment:
Nit: We can configure checkstyle in an external XML, by which we can avoid flooding the `pom.xml`.

Reply:
i can try

Reply:
You can put into a xml, i don't really like because makes maven a bit more complexer than just inline. The Checker is executed in each sub project and from their you have to find the way to a central config.

Reply:
If that causes too much config overhead, you can also leave it in `pom.xml`